### DISCUSSION TITLE
Why can't I clone any repositories using Git at the command line?

### DISCUSSION THREAD

**Discussion by Rod-at-DOH** [community/community#135144] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Suddenly Git (version 2.46) will **NOT** let me clone any repo, when I try cloning it at the command line. (Windows 11, PowerShell 7.) I don't know if this is an issue with Git, or an issue with GitHub, so I'm posting this question here. When I try cloning a repo out of one of our Enterprise Organizations, I get the following:

> Unhandled Exception: System.MissingMethodException: Method not found: 'System.Func`2<System.IServiceProvider,System.Object> Avalonia.Markup.Xaml.XamlIl.Runtime.XamlIlRuntimeHelpers.DeferredTransformationFactoryV2(System.Func`2<System.IServiceProvider,System.Object>, System.IServiceProvider)'.
   at Avalonia.Themes.Fluent.FluentTheme.!XamlIlPopulate(IServiceProvider , FluentTheme )
   at Avalonia.Themes.Fluent.FluentTheme.!XamlIlPopulateTrampoline(IServiceProvider , FluentTheme )
   at Avalonia.Themes.Fluent.FluentTheme..ctor(IServiceProvider sp)
   at GitCredentialManager.UI.AvaloniaApp.!XamlIlPopulate(IServiceProvider , AvaloniaApp )
   at GitCredentialManager.UI.AvaloniaApp.!XamlIlPopulateTrampoline(AvaloniaApp )
   at GitCredentialManager.UI.AvaloniaApp.Initialize()
   at Avalonia.AppBuilder.SetupUnsafe()
   at Avalonia.AppBuilder.Setup()
   at Avalonia.AppBuilder.SetupWithoutStarting()
   at GitCredentialManager.UI.AvaloniaUi.<>c__DisplayClass8_1.<ShowWindowAsync>b__1(CancellationToken appCancelToken)
   at GitCredentialManager.UI.Dispatcher.DispatcherJob.Execute(CancellationToken ct)
   at GitCredentialManager.UI.Dispatcher.DispatcherJobQueue.Run()
   at GitCredentialManager.UI.Dispatcher.Run()
   at GitCredentialManager.Program.Main(String[] args)

It then asks for my username and password to GitHub, I presume. I enter both and then I get this:

> remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/OUR_ORGANIZATION/XXXXX.git/'

Why in heck does it ask for a username and password, when it then turns around and tells me it won't accept my username and password, and hasn't accepted usernames and passwords since August 13, 2021???

I can clone the repo using Visual Studio, but I prefer to do this at the command line. Why can't I do this now, when I've been able to clone any repo I have access to for years?

**Comment by dane** [comment]
When `git` asks for a username and password, it is actually asking for your GitHub username and a developer token. Create one [in your settings](https://github.com/settings/tokens) scoped to repos or the specific repos you want to work on.

**Reply by Rod-at-DOH** [reply]
Really? That's what it's asking for? Then how do I specify a token, when it's asking for a username and password? And what would have prompted this change in requirements? I've asked because I've not had to do this before. I'd like to know what's changed?

**Reply by dane** [reply, marked as answer]
`git` is written with a username and password in mind and GitHub is limited to what `git` will communicate to the user. I know, it's misleading. They made this change because a password it scoped to everything in your GitHub account, but tokens can be scoped to read/write access on specific resources and even finer grained permissions (eg: read/write to a single repository).

To get a token navigate to: Settings > Developer Settings > Personal access tokens. To get you going, I'd suggest making a classic token with the "repo" checkbox selected. Setting the expiration to "No expiration" will be most similar to you using your original password, but it's not recommended.

Why would you want to use a fine-grained token? If you want to give a bot access to a read or modify a single repo. It can also be used a forced focus function. Creating a token that only lasts 7 days and is scoped to one or two repos that you want to work on, would prevent you from getting distracted and tinkering with projects outside of your plans.

I hope this helps!

**Reply by Rod-at-DOH** [reply]
This does help. Thank you!
